frontend/src/utils/numbers.ts

```diff
@@ -1,7 +1,31 @@
 /* Copyright 2024 Marimo. All rights reserved. */
+import { format } from "mathjs";
+
 export function prettyNumber(value: number): string {
   return value.toLocaleString(undefined, {
     minimumFractionDigits: 0,
     maximumFractionDigits: 2,
   });
 }
+
+export function prettyScientificNumber(value: number): string {
+  // Handle special cases first
+  if (value === 0) {
+    return "0";
+  } // Avoid displaying -0
+  if (Number.isNaN(value)) {
+    return "NaN";
+  }
+  if (!Number.isFinite(value)) {
+    return value > 0 ? "Infinity" : "-Infinity";
+  }
+
+  return Math.trunc(value) === 0
+    ? // No integer part, use scientific notation
+      format(value, { notation: "auto", precision: 2 })
```

Comment:
the browser has this built in now. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#scientific

would that work instead and avoid the extra dep? we can lowercase the `e` aswell to match this formatting. 


```ts
new Intl.NumberFormat("en-US", {
  notation: "scientific",
}).format(987654321);
```

Reply:
I think there is a bug for this, the slider would disappear after some dragging, so I retain to `mathjs`